Refactored to support netstandard2.0 and net4.5

mitch-tofi opened the pull request:
Hey @Redth,

Firstly <3 the project really helps!, however it didn't support .net core (namely 2.0),

I have forked it and currently have a nuget package live (PushSharp.Core), so I can use it.

Creating this PR in case you want to pull it into your code (I did introduce breaking changes, primarily around the removal of the obsolete properties in GCM and incorrectly named class)

Effectively project now supports net45 and netstandard2.0.

if you do pull it in, ill remove more nuget package to avoid confusion,

Cheers,

Mitch

mitch-tofi committed:
Ported across and created package called PushSharp.Core

Jose7777 commented:
Hello Mitch,

Does Pushsharp.Core include other PRs that fixes a lot of problemas, like #753 or #774, or only includes support por Core 2.0?
Is production ready?

I'm planning to migrate to .net core 2.0, and this is really helpful.

Thank you very much
Jose

mitch-tofi commented:
HI Jose,

This was just a port of the existing code base, so I haven't addressed any of the open issues i'm afraid,

I've been using it for a while now and seems to be working fine for me so far.

Mitch

Caezary committed:
add APNS and GCM proxy support (#1)

* Update PushShart using a Proxy when connecting to Google Cloud Messaging (GCM)

* Options for Proxy Usage

Adds properties and methods to use an HTTP-proxy in your connection, either with default network credentials or specified credentials.

* Connection via proxy

A method to tunnel the TCP-connection via a pre-configured HTTP-Proxy has been added. If Configuration.UseProxy is set to true, a CONNECT-request is sent using the proxy adress specified. The socket is then extracted from the stream of this request's response (using the System.Reflection-Library) and used to create a TcpClient-object. All traffic sent through this TcpClient-object will pass through the proxy used to send the initial CONNECT-Request.

NOTE:
The method 'getClientViaHTTPProxy' uses code that was not written by the author of this commit. The original code can be found here:
https://web.archive.org/web/20160317134733/https://nitormobiledevelopment.wordpress.com/2013/08/13/push-sharp-using-proxy/

* Update ApnsConnection.cs

* Update ApnsConfiguration.cs

* apns wit proxy withou socket hijack

* Handling proxy connection errors on APNS

* Using System.Net missing in GcmConfiguration

* Http Proxy with Basic authentication without WebRequest in APNS

* System.Net was missing in GcmConfiguration

* Using http proxy with basic auth (not using socket hijacking) in APNS

* New class ProxyHelper to permit proxy connect in any place using a TcpClient

* Using ProxyHelper in ApnsConnection

* Supporting proxy in ApnsFeedbackService.

* Changes in ApnsFeedbackService to return the tokens list intead of raising events

* add proxy usage section to readme

rvidis committed:
Update readme - syntax error (#2)

rmidyk-cashplus committed:
APNS - add support of TLS1.1 and TLS1.2 (#3)

KANekT committed:
Update GcmConfiguration.cs (#4)

https://gcm-http.googleapis.com/gcm/send

is Error=DeprecatedEndpoint
